api/src/opentrons/protocol_engine/state/modules.py

```diff
@@ -182,41 +191,88 @@ def _add_module_substate(
         elif ModuleModel.is_heater_shaker_module_model(model):
             self._state.substate_by_module_id[module_id] = HeaterShakerModuleSubState(
                 module_id=HeaterShakerModuleId(module_id),
-                plate_target_temperature=None,
+                is_labware_latch_closed=True
+                if (live_data and live_data["labwareLatchStatus"] == "idle_closed")
+                else False,
+                is_plate_shaking=True
+                if (live_data and live_data["targetSpeed"])
+                else False,
```

Comment:
`True if ... else False` can be simplified to just the condition. Also, this is pretty hard to read without parentheses

```suggestion
                is_labware_latch_closed=(
                    live_data is not None
                    and live_data["labwareLatchStatus"] == "idle_closed"
                ),
                is_plate_shaking=(
                    live_data is not None and live_data["targetSpeed"] is not None
                ),
```